src/chainstate/stacks/db/blocks.rs

```diff
@@ -4130,28 +4262,131 @@ impl StacksChainState {
         return true;
     }
 
-    /// Find and process the next staging block.
-    /// Return the next chain tip if we processed this block, or None if we couldn't.
-    /// Return a poison microblock transaction payload if the microblock stream contains a
-    /// deliberate miner fork.
-    ///
-    /// Occurs as a single, atomic transaction against the (marf'ed) headers database and
-    /// (un-marf'ed) staging block database, as well as against the chunk store.
-    pub fn process_next_staging_block(
-        &mut self,
-        sort_tx: &mut SortitionHandleTx,
-    ) -> Result<(Option<StacksEpochReceipt>, Option<TransactionPayload>), Error> {
-        let (mut chainstate_tx, clarity_instance) = self.chainstate_tx_begin()?;
-
-        let blocks_path = chainstate_tx.blocks_tx.get_blocks_path().clone();
-
-        // this is a transaction against both the headers and staging blocks databases!
-        let (mut next_microblocks, next_staging_block) =
-            match StacksChainState::find_next_staging_block(
-                &mut chainstate_tx.blocks_tx,
-                &blocks_path,
-                &chainstate_tx.headers_tx,
-                sort_tx,
+    /// Get the parent header info for a block we're processing, if it's known.
+    /// The header info will be pulled from the headers DB, so this method only succeeds if the
+    /// parent block has been processed.
+    /// If it's not known, return None.
+    fn get_parent_header_info<'a>(
```

Comment:
Functions where there's only one lifetime parameter and that lifetime parameter isn't in the return type don't need to specify the lifetime. So this function (and a lot of others in this file) could be specified like:

`fn get_parent_header_info(chainstate_tx: &mut ChainstateTx, next_staging_block: ...`

Reply:
Addressed in ee12847